pyiron/atomistics/structure/factory.py

```diff
@@ -0,0 +1,333 @@
+                                      directions.
+            **kwargs: Additional, arguments you would normally pass to the structure generator like 'a', 'b',
+            'orthogonal' etc.
+
+        Returns:
+            pyiron.atomistics.structure.atoms.Atoms instance: Required surface
+
+        """
+        # https://gitlab.com/ase/ase/blob/master/ase/lattice/surface.py
+        if pbc is None:
+            pbc = True
+        s.publication_add(publication_ase())
+        for surface_class in [
+            add_adsorbate,
+            add_vacuum,
+            bcc100,
+            bcc110,
+            bcc111,
+            diamond100,
+            diamond111,
+            fcc100,
+            fcc110,
+            fcc111,
+            fcc211,
+            hcp0001,
+            hcp10m10,
+            mx2,
+            hcp0001_root,
+            fcc111_root,
+            bcc111_root,
+            root_surface,
+            root_surface_analysis,
+            ase_surf,
+        ]:
+            if surface_type == surface_class.__name__:
+                surface_type = surface_class
+                break
+        if isinstance(surface_type, types.FunctionType):
+            if center:
+                surface = surface_type(
+                    symbol=element, size=size, vacuum=vacuum, **kwargs
+                )
+            else:
+                surface = surface_type(symbol=element, size=size, **kwargs)
+                z_max = np.max(surface.positions[:, 2])
+                surface.cell[2, 2] = z_max + vacuum
+            surface.pbc = pbc
+            return ase_to_pyiron(surface)
+        else:
+            return None
+
+    @staticmethod
+    def hkl_surface(lattice, hkl, layers, vacuum=1.0, center=False, pbc=True):
+        """
+        Use ase.build.surface to build a surface with surface normal (hkl).
+
+        Args:
+            lattice (pyiron.atomistics.structure.atoms.Atoms/str): bulk Atoms
+                instance or str, e.g. "Fe", from which to build the surface
+            hkl (list): miller indices of surface to be created
+            layers (int): # of atomic layers in the surface
+            vacuum (float): vacuum spacing
+            center (bool): shift all positions to center the surface
+                in the cell
+
+        Returns:
+            pyiron.atomistics.structure.atoms.Atoms instance: Required surface
+        """
+        # https://gitlab.com/ase/ase/blob/master/ase/lattice/surface.py
+        s.publication_add(publication_ase())
+
+        surface = ase_surf(lattice, hkl, layers)
+        z_max = np.max(surface.positions[:, 2])
+        surface.cell[2, 2] = z_max + vacuum
+        if center:
+            surface.positions += 0.5 * surface.cell[2] - [0, 0, z_max/2]
+        surface.pbc = pbc
+        return ase_to_pyiron(surface)
+
+    @staticmethod
+    def structure(element, bravais_basis, lattice_constant):
```

Comment:
```suggestion
    def crystal(element, bravais_basis, lattice_constant):
```
The CrystalStructure class is the original pyiron structure generator but I have the feeling most of us use the ASE structure generator these days so I would remove this function. But maybe we should move that discussion to a separate pull request.

Reply:
@jan-janssen I'm ok with this suggestion for this PR, but actually I was toying around with the idea of modifying `StructureFactory.__call__` so that we can simply say `pr.create.structure('Al', 'fcc', 4.04)` while still being able to say `pr.create.structure.surface(...)` etc.

In fact, if I were being *really* bold, I would want `pr.create.structure(...)` to be equivalent to our current `pr.create_ase_bulk(...)`, and `pr.create.bulk(...)` to be equivalent to our current `pr.create_structure`.

Thoughts? I agree it can be a separate PR though, and in the meantime will commit your suggestions.

Reply:
I like the idea with the `__call__` I was not even aware of this and I am happy to match it with `pr.create.bulk()` but I would still prefer to remove the old `Crystalstructure` class completely unless somebody else is using it regularly. Again this should be a separate pull request.

Reply:
Agreed on both counts.